src/spdx/parser/rdf/graph_parsing_functions.py

```diff
@@ -0,0 +1,113 @@
+from spdx.casing_tools import camel_case_to_snake_case
+
+
+def parse_literal(logger: Logger, graph: Graph, subject: Node, predicate: Node,
+                  parsing_method: Callable = lambda x: x.strip(), default: Any = None):
+    value = get_unique_value(logger, graph, subject, predicate, default)
+    if not value:
+        return default
+    return apply_parsing_method_or_log_error(logger, value, parsing_method, default)
+
+
+def apply_parsing_method_or_log_error(logger: Logger, value: Any, parsing_method: Callable = lambda x: x.strip(),
+                                      default: Any = None):
+    try:
+        return parsing_method(value)
+    except SPDXParsingError as err:
+        logger.extend(err.get_messages())
+    except (TypeError, ValueError) as err:
+        logger.append(err.args[0])
+    return default
+
+
+def parse_literal_or_no_assertion_or_none(logger: Logger, graph: Graph, subject: Node, predicate: Node,
+                                          parsing_method: Callable = lambda x: x.strip(), default: Any = None):
+    value = get_unique_value(logger, graph, subject, predicate, default)
+    if not value:
+        return default
+    if value == SPDX_NAMESPACE.noassertion or value.toPython() == SPDX_NO_ASSERTION_STRING:
+        return SpdxNoAssertion()
+    if value == SPDX_NAMESPACE.none or value.toPython() == SPDX_NONE_STRING:
+        return SpdxNone()
+    return apply_parsing_method_or_log_error(logger, value, parsing_method, default)
+
+
+def get_correct_typed_value(logger: Logger, value: Any, parsing_method: Callable = lambda x: x.strip(),
+                            default: Any = None):
+    if not value:
+        return default
+    if value == SPDX_NAMESPACE.noassertion or value.toPython() == SPDX_NO_ASSERTION_STRING:
+        return SpdxNoAssertion()
+    if value == SPDX_NAMESPACE.none or value.toPython() == SPDX_NONE_STRING:
+        return SpdxNone()
+    return apply_parsing_method_or_log_error(logger, value, parsing_method, default)
+
+
+def parse_literal_or_no_assertion(logger: Logger, graph: Graph, subject: Node, predicate: Node,
+                                  parsing_method: Callable = str, default: Any = None):
+    value = get_unique_value(logger, graph, subject, predicate, default)
+    if not value:
+        return default
+    if value == SPDX_NAMESPACE.noassertion or value.toPython() == SPDX_NO_ASSERTION_STRING:
+        return SpdxNoAssertion()
+    return apply_parsing_method_or_log_error(logger, value, parsing_method, default)
+
+
+def get_unique_value(logger: Logger, graph: Graph, subject: Node, predicate: Node, default: Any) -> Any:
+    try:
+        value = graph.value(subject=subject, predicate=predicate, default=default, any=False)
+        return value
+    except UniquenessError:
+        logger.append(f"Multiple values for unique value {predicate} found.")
+        return default
+
+
+def parse_enum_value(enum_str: str, enum_class: Type[Enum], prefix: str) -> Enum:
```

Comment:
ChecksumAlgorithm needs a special conversion because we need to insert an underscore for the `BLAKE2B` algorithms. FileType doesn't need the conversion from camel case to snake case but as this function would not make a difference, we can use `parse_enumvalue` also for FileType for consistency.